src/realm/status.hpp

```diff
@@ -41,22 +41,13 @@ class REALM_NODISCARD Status {
      */
     template <typename Reason, std::enable_if_t<std::is_constructible_v<std::string_view, Reason>, int> = 0>
     Status(ErrorCodes::Error code, Reason&& reason)
-        : m_error(ErrorInfo::create(code, std::string_view{reason}))
+        : m_error(ErrorInfo::create(code, std::string{std::string_view{reason}}))
     {
     }
 
-    template <typename Reason, std::enable_if_t<std::is_constructible_v<std::string_view, Reason>, int> = 0>
-    Status(std::error_code code, Reason&& reason)
-    {
-        if (code) {
-            m_error = ErrorInfo::create(ErrorCodes::SystemError, std::string_view{reason});
-            m_error->m_std_error_code = code;
-        }
-    }
-
-    std::error_code get_std_error_code() const
+    Status(ErrorCodes::Error code, std::string&& reason)
```

Comment:
Do we want/need both constructors?

Reply:
Yeah, so the first constructor makes a copy of a string. This constructor is if you've just created a Status with a new string via util::format() or something. So this is just a little optimization to avoid an allocation/copy.